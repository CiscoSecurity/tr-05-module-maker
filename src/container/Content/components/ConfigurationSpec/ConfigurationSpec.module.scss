@import "src/globals/constants/colors";

.addNewButton {
  flex: 1;
  height: 28px;
  min-width: 75px;
  color: $white-color;
  background-color: $blue-background;
  border-radius: 3px;
  border: 0;
  box-shadow: none;
  font-weight: 400;
  line-height: 1.5;
  text-overflow: ellipsis;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  white-space: nowrap;
  margin: 4px;
}

.confSpecWrapper {
  flex: 1;
  display: flex;
  flex-direction: row;
  border-radius: 3px;
  border: 1px solid $border-color;
  padding: 4px;
  margin-bottom: 10px;
}

.row {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.column {
  flex: 1;
  display: flex;
  flex-direction: column;
  margin-left: 4px;
}

.inputDiv {
  flex: 1;
  display: flex;
  align-items: center;
}

.customInput {
  flex: 1;
  margin: 4px;
}

.checkbox {
  margin-top: 4px;
  margin-bottom: 4px;
}

.closeIcon {
  fill: $close-icon-fill;
  width: 15px;
  height: 15px;
}

.optionsIconWrapper {
  flex: 1;
  align-self: flex-end;
}

.optionsWrapper {
  flex: 1;
  display: flex;
  flex-direction: column;
  border-radius: 3px;
  border: 1px solid $border-color;
  padding: 4px;
}