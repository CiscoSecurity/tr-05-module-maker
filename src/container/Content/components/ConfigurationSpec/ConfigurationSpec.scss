@import "src/globals/constants/colors";

.confSpecRow {
  display: flex;
  align-items: center;
  justify-content: space-between;

  button {
    flex: 1;
    height: 28px;
    min-width: 75px;
    color: $white-color;
    background-color: $blue-background;
    border-radius: 3px;
    border: 0;
    box-shadow: none;
    font-weight: 400;
    line-height: 1.5;
    text-overflow: ellipsis;
    text-align: center;
    vertical-align: middle;
    cursor: pointer;
    white-space: nowrap;
    margin: 4px;
  }
}

