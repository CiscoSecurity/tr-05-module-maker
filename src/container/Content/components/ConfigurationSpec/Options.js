import React from "react";
import styles from "./ConfigurationSpec.module.scss";
import Icons from "../../../../globals/icons/sprite.svg";


export default class Options extends React.Component {
    render() {
        return (
            <div className={styles.optionsWrapper}>
                <div className={styles.optionsIconWrapper}>
                    <svg className={styles.closeIcon}>
                        <use xlinkHref={`${Icons}#icon-small-x-close`}/>
                    </svg>
                </div>
                <div className={styles.inputDiv}>
                    value
                    <input type="text" name="option" className={styles.customInput}/>
                </div>
                <div className={styles.inputDiv}>
                    label
                    <input type="text" name="option" className={styles.customInput}/>
                </div>
            </div>
        )
    }
}
