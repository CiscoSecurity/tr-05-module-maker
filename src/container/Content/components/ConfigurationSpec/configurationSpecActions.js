import {
    ADD_CONF_SPEC, ADD_OPTIONS,
    DELETE_ALL_OPTIONS,
    DELETE_CONF_SPEC,
    DELETE_OPTION,
    UPDATE_CONF_SPEC, UPDATE_OPTION
} from "globals/constants/types";

export const addConfSpec = () => ({
        type: ADD_CONF_SPEC,
        payload:   {
            "key": "",
            "type": "",
            "label": "",
            "tooltip": "",
            "subtype": "",
            "required": false,
            "group": "",
            "options": [],
            "id": new Date().getTime()
        }
})

export const deleteConfSpec = (id) => ({
        type: DELETE_CONF_SPEC,
        payload: {conf_spec_id: id}
})

export const updateConfSpec = (id, pair) => ({
        type: UPDATE_CONF_SPEC,
        payload: {
            conf_spec_id: id,
            pair: pair
        }
})

export const addOptions = (id) => {
    return {
        type: ADD_OPTIONS,
        payload:
            {
                options: {
                    value: "", "label": "",
                    id: new Date().getTime()
                },
                conf_spec_id: id
            }
    }
}

export const deleteAllOptions = (id) => ({
        type: DELETE_ALL_OPTIONS,
        payload: {conf_spec_id: id}
})

export const deleteOption = (option_id, conf_spec_id) => ({
        type: DELETE_OPTION,
        payload: {option_id: option_id, conf_spec_id: conf_spec_id}
})

export const updateOption = (option_id, conf_spec_id, pair) => ({
        type: UPDATE_OPTION,
        payload: {option_id: option_id, conf_spec_id: conf_spec_id, pair: pair}
})
