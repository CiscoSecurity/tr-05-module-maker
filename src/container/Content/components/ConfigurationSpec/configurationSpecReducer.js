import {
    ADD_CONF_SPEC,
    ADD_OPTIONS,
    DELETE_ALL_OPTIONS,
    DELETE_CONF_SPEC,
    DELETE_OPTION,
    UPDATE_CONF_SPEC,
    UPDATE_OPTION
} from "globals/constants/types"


export const configurationSpecReducer = (state = [], action) => {
    switch (action.type) {
        case ADD_CONF_SPEC:
            return state.concat([action.payload])
        case ADD_OPTIONS:
            let item = state.filter((spec) => spec.id === action.payload.conf_spec_id);
            let elementIndex = state.indexOf(item[0]);
            return state.map((item, index) => {
                    if (index !== elementIndex) {
                        return item
                    }
                    return {
                        ...item,
                        "options": item.options.concat(action.payload.options)
                    }
                }
            )

        case DELETE_CONF_SPEC:
            return state.filter((spec) => spec.id !== action.payload);

        case DELETE_ALL_OPTIONS:
            let elemToDelete = state.filter((spec) => spec.id === action.payload);
            let indexToDelete = state.indexOf(elemToDelete[0]);
            return state.map((item, index) => {
                    if (index !== indexToDelete) {
                        return item
                    }
                    return {
                        ...item,
                        "options": []
                    }
                }
            )
        case DELETE_OPTION:
            let spec = state.filter((spec) => spec.id === action.payload.conf_spec_id);
            let indexOfSpec = state.indexOf(spec[0]);
            return state.map((item, index) => {
                    if (index !== indexOfSpec) {
                        return item
                    }
                    return {
                        ...item,
                        "options": item.options.filter(el => el.id !== action.payload.option_id)
                    }
                }
            )

        case UPDATE_CONF_SPEC:
            let itemToUpdate = state.filter((spec) => spec.id === action.payload.id);
            let indexToUpdate = state.indexOf(itemToUpdate[0]);
            return state.map((item, index) => {
                    if (index !== indexToUpdate) {
                        return item
                    }
                    return {
                        ...item,
                        [action.payload.pair.name]: action.payload.pair.value
                    }
                }
            )

        case UPDATE_OPTION:
            let optionToUpdate = state.filter((spec) => spec.id === action.payload.conf_spec_id);
            let indexOfOption = state.indexOf(optionToUpdate[0]); //ToDo --> indexOfOption --> option_id
            return state.map((item, index) => {
                    if (index !== indexOfOption) {
                        return item
                    }
                    return {
                        ...item,
                        "options": item.options.map((option, ind) => {
                                if (option.id !== action.payload.option_id) {
                                    return option
                                }
                                return {
                                    ...option,
                                    [action.payload.pair.name]: action.payload.pair.value
                                }
                            }
                        )
                    }
                }
            )

        default:
            return state
    }
}
