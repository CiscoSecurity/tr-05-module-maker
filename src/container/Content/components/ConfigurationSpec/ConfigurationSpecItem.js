import React from "react";
import styles from "./ConfigurationSpec.module.scss"
import * as Constants from "../../../../globals/constants/constants";
import Icons from "../../../../globals/icons/sprite.svg";
import Options from "./Options";


export default class ConfigurationSpecItem extends React.Component {

    render() {
        return (
            <div className={styles.confSpecWrapper}>
                <div className={styles.column}>
                    <div>
                        <div className={styles.inputDiv}>
                            key
                            <input
                                type="text"
                                name="key"
                                className={styles.customInput}
                                required
                                list="key-input-list"
                            />
                        </div>
                        <datalist id="key-input-list">
                            {
                                Constants.KEY_DATALIST.map(
                                    option => {
                                        return <option>{option}</option>
                                    }
                                )
                            }
                        </datalist>
                        <div className={styles.inputDiv}>
                            type
                            <select name="type" className={styles.customInput} defaultValue={'DEFAULT'}>
                                <option value="DEFAULT" disabled hidden>{Constants.SELECT_PLACEHOLDER}</option>
                                {
                                    Constants.TYPE_OPTIONS.map(
                                        option => {
                                            return <option>{option}</option>
                                        }
                                    )
                                }
                            </select>
                        </div>
                        <div className={styles.inputDiv}>
                            label
                            <input type="text" name="label" className={styles.customInput} required/>
                        </div>
                    </div>
                </div>
                <div className={styles.column}>
                    <div>
                        <div className={styles.inputDiv}>
                            tooltip
                            <input type="text" name="tooltip" className={styles.customInput}/>
                        </div>
                        <div className={styles.inputDiv}>
                            subtype
                            <input type="text" name="subtype" className={styles.customInput}/>
                        </div>
                        <div className={styles.inputDiv}>
                            group
                            <input type="text" name="group" className={styles.customInput}/>
                        </div>
                        <div className={styles.checkbox}>
                            required
                            <input type="checkbox" name="required"/>
                        </div>
                    </div>
                </div>
                <div className={styles.column}>
                    <div className={styles.row}>
                        <div>
                            Options
                            <button className={styles.addNewButton}>Add new</button>
                        </div>
                        <div>
                            <svg className={styles.closeIcon}>
                                <use xlinkHref={`${Icons}#icon-small-x-close`}/>
                            </svg>
                        </div>
                    </div>
                    <Options/>
                </div>
            </div>
        )
    }
}
