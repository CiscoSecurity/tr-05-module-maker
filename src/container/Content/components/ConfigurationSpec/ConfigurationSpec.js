import React from "react";
import styles from "./ConfigurationSpec.module.scss"
import ConfigurationSpecItem from "./ConfigurationSpecItem"
import { addConfSpec } from "../../../../redux/actions";
import { connect } from "react-redux";

class ConfigurationSpec extends React.Component {
    render() {
        return (
            <div>
                <div className={styles.row}>
                    <div>
                    Configuration Spec
                    <button className={styles.addNewButton} onClick={this.props.addConfSpec}> Add new </button>
                    </div>
                </div>
                {
                    this.props.syncConfSpec.map(
                    () => (<ConfigurationSpecItem />)
                    )
                }

            </div>
        )
    }
}

const mapDispatchToProps = {
    addConfSpec,
}
const mapStateToProps = state => {
    return {
        syncConfSpec: state.json.configuration_spec
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ConfigurationSpec);
