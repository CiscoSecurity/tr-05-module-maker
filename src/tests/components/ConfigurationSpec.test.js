import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import ConfigurationSpec from "../../container/Content/components/ConfigurationSpec/ConfigurationSpec";
import React from "react";
import configureStore from 'redux-mock-store';
import toJson from "enzyme-to-json";
import * as mocks from "../mocks/mockPayloads"


const mockStore = configureStore([]);


describe('ConfigurationSpec', () => {
    it('should render ConfigurationSpec without ConfigurationSpecItems', () => {
        const store = mockStore({
            configuration_spec: [],
        });

        const wrapper = mount(
            <Provider store={store}>
                <ConfigurationSpec/>
            </Provider>);
        expect(wrapper.find('ConfigurationSpecItem')).toHaveLength(0);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should render ConfigurationSpec with several ConfigurationSpecItems', () => {
        const store = mockStore({
            configuration_spec: [
                mocks.EMPTY_CONF_SPEC_MOCK,
                mocks.FILLED_CONF_SPEC_MOCK
            ]
        });

        const wrapper = mount(
            <Provider store={store}>
                <ConfigurationSpec/>
            </Provider>);
        expect(wrapper.find('ConfigurationSpecItem')).toHaveLength(2);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
