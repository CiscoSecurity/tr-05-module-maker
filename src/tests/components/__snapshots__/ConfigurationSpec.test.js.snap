// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationSpec should render ConfigurationSpec with several ConfigurationSpecItems 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ConfigurationSpec)>
    <ConfigurationSpec
      addConfSpec={[Function]}
      syncConfSpec={
        Array [
          Object {
            "group": "",
            "id": "00000000-0000-0000-0000-000000000000",
            "key": "",
            "label": "",
            "options": Array [],
            "required": false,
            "subtype": "",
            "tooltip": "",
            "type": "",
          },
          Object {
            "id": "00000000-0000-0000-0000-000000000002",
            "key": "url",
            "label": "URL",
            "options": Array [
              Object {
                "id": "00000000-0000-0000-0000-000000000001",
                "label": "https://example.com/",
                "value": "https://example.com/",
              },
            ],
            "required": true,
            "tooltip": "The base URL of the Serverless Relay",
            "type": "string",
          },
        ]
      }
    >
      <div>
        <div
          className="confSpecRow"
        >
          <div>
            Configuration Spec
            <button
              onClick={[Function]}
              type="button"
            >
              + Add
            </button>
          </div>
        </div>
        <Connect(ConfigurationSpecItem)
          key="00000000-0000-0000-0000-000000000000"
          role="00000000-0000-0000-0000-000000000000"
          syncConfSpecItem={
            Object {
              "group": "",
              "id": "00000000-0000-0000-0000-000000000000",
              "key": "",
              "label": "",
              "options": Array [],
              "required": false,
              "subtype": "",
              "tooltip": "",
              "type": "",
            }
          }
        >
          <ConfigurationSpecItem
            addOptions={[Function]}
            deleteAllOptions={[Function]}
            deleteConfSpec={[Function]}
            role="00000000-0000-0000-0000-000000000000"
            syncConfSpec={
              Array [
                Object {
                  "group": "",
                  "id": "00000000-0000-0000-0000-000000000000",
                  "key": "",
                  "label": "",
                  "options": Array [],
                  "required": false,
                  "subtype": "",
                  "tooltip": "",
                  "type": "",
                },
                Object {
                  "id": "00000000-0000-0000-0000-000000000002",
                  "key": "url",
                  "label": "URL",
                  "options": Array [
                    Object {
                      "id": "00000000-0000-0000-0000-000000000001",
                      "label": "https://example.com/",
                      "value": "https://example.com/",
                    },
                  ],
                  "required": true,
                  "tooltip": "The base URL of the Serverless Relay",
                  "type": "string",
                },
              ]
            }
            syncConfSpecItem={
              Object {
                "group": "",
                "id": "00000000-0000-0000-0000-000000000000",
                "key": "",
                "label": "",
                "options": Array [],
                "required": false,
                "subtype": "",
                "tooltip": "",
                "type": "",
              }
            }
            updateConfSpec={[Function]}
          >
            <div
              className="confSpecWrapper"
            >
              <div
                className="column"
              >
                <div
                  className="inputDiv"
                >
                  <label>
                    key
                  </label>
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                  <input
                    className="customInput"
                    list="key-input-list"
                    name="key"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
                <datalist
                  id="key-input-list"
                >
                  <option
                    key="basic-auth-password 00000000-0000-0000-0000-000000000000"
                  >
                    basic-auth-password
                  </option>
                  <option
                    key="basic-auth-user 00000000-0000-0000-0000-000000000000"
                  >
                    basic-auth-user
                  </option>
                  <option
                    key="token 00000000-0000-0000-0000-000000000000"
                  >
                    token
                  </option>
                  <option
                    key="url 00000000-0000-0000-0000-000000000000"
                  >
                    url
                  </option>
                  <option
                    key="user 00000000-0000-0000-0000-000000000000"
                  >
                    user
                  </option>
                </datalist>
                <div
                  className="inputDiv"
                >
                  <label>
                    type
                  </label>
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                  <select
                    className="customInput"
                    name="type"
                    onChange={[Function]}
                    required={true}
                    value=""
                  >
                    <option
                      disabled={true}
                      hidden={true}
                      value=""
                    >
                      Select your option
                    </option>
                    <option
                      key="api_key"
                    >
                      api_key
                    </option>
                    <option
                      key="boolean"
                    >
                      boolean
                    </option>
                    <option
                      key="device"
                    >
                      device
                    </option>
                    <option
                      key="integer"
                    >
                      integer
                    </option>
                    <option
                      key="options"
                    >
                      options
                    </option>
                    <option
                      key="password"
                    >
                      password
                    </option>
                    <option
                      key="string"
                    >
                      string
                    </option>
                  </select>
                </div>
                <div
                  className="inputDiv"
                >
                  <label>
                    label
                  </label>
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                  <input
                    className="customInput"
                    name="label"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="inputDiv"
                >
                  <label>
                    tooltip
                  </label>
                  <input
                    className="customInput"
                    name="tooltip"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="inputDiv"
                >
                  <label>
                    subtype
                  </label>
                  <input
                    className="customInput"
                    name="subtype"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="inputDiv"
                >
                  <label>
                    group
                  </label>
                  <input
                    className="customInput"
                    name="group"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="checkboxDiv"
                >
                  <label>
                    required
                  </label>
                  <input
                    checked={false}
                    name="required"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </div>
              <div>
                <svg
                  className="closeIcon"
                  onClick={[Function]}
                >
                  <use
                    xlinkHref="sprite.svg#icon-small-x-close"
                  />
                </svg>
              </div>
            </div>
          </ConfigurationSpecItem>
        </Connect(ConfigurationSpecItem)>
        <Connect(ConfigurationSpecItem)
          key="00000000-0000-0000-0000-000000000002"
          role="00000000-0000-0000-0000-000000000002"
          syncConfSpecItem={
            Object {
              "id": "00000000-0000-0000-0000-000000000002",
              "key": "url",
              "label": "URL",
              "options": Array [
                Object {
                  "id": "00000000-0000-0000-0000-000000000001",
                  "label": "https://example.com/",
                  "value": "https://example.com/",
                },
              ],
              "required": true,
              "tooltip": "The base URL of the Serverless Relay",
              "type": "string",
            }
          }
        >
          <ConfigurationSpecItem
            addOptions={[Function]}
            deleteAllOptions={[Function]}
            deleteConfSpec={[Function]}
            role="00000000-0000-0000-0000-000000000002"
            syncConfSpec={
              Array [
                Object {
                  "group": "",
                  "id": "00000000-0000-0000-0000-000000000000",
                  "key": "",
                  "label": "",
                  "options": Array [],
                  "required": false,
                  "subtype": "",
                  "tooltip": "",
                  "type": "",
                },
                Object {
                  "id": "00000000-0000-0000-0000-000000000002",
                  "key": "url",
                  "label": "URL",
                  "options": Array [
                    Object {
                      "id": "00000000-0000-0000-0000-000000000001",
                      "label": "https://example.com/",
                      "value": "https://example.com/",
                    },
                  ],
                  "required": true,
                  "tooltip": "The base URL of the Serverless Relay",
                  "type": "string",
                },
              ]
            }
            syncConfSpecItem={
              Object {
                "id": "00000000-0000-0000-0000-000000000002",
                "key": "url",
                "label": "URL",
                "options": Array [
                  Object {
                    "id": "00000000-0000-0000-0000-000000000001",
                    "label": "https://example.com/",
                    "value": "https://example.com/",
                  },
                ],
                "required": true,
                "tooltip": "The base URL of the Serverless Relay",
                "type": "string",
              }
            }
            updateConfSpec={[Function]}
          >
            <div
              className="confSpecWrapper"
            >
              <div
                className="column"
              >
                <div
                  className="inputDiv"
                >
                  <label>
                    key
                  </label>
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                  <input
                    className="customInput"
                    list="key-input-list"
                    name="key"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="url"
                  />
                </div>
                <datalist
                  id="key-input-list"
                >
                  <option
                    key="basic-auth-password 00000000-0000-0000-0000-000000000002"
                  >
                    basic-auth-password
                  </option>
                  <option
                    key="basic-auth-user 00000000-0000-0000-0000-000000000002"
                  >
                    basic-auth-user
                  </option>
                  <option
                    key="token 00000000-0000-0000-0000-000000000002"
                  >
                    token
                  </option>
                  <option
                    key="url 00000000-0000-0000-0000-000000000002"
                  >
                    url
                  </option>
                  <option
                    key="user 00000000-0000-0000-0000-000000000002"
                  >
                    user
                  </option>
                </datalist>
                <div
                  className="inputDiv"
                >
                  <label>
                    type
                  </label>
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                  <select
                    className="customInput"
                    name="type"
                    onChange={[Function]}
                    required={true}
                    value="string"
                  >
                    <option
                      disabled={true}
                      hidden={true}
                      value=""
                    >
                      Select your option
                    </option>
                    <option
                      key="api_key"
                    >
                      api_key
                    </option>
                    <option
                      key="boolean"
                    >
                      boolean
                    </option>
                    <option
                      key="device"
                    >
                      device
                    </option>
                    <option
                      key="integer"
                    >
                      integer
                    </option>
                    <option
                      key="options"
                    >
                      options
                    </option>
                    <option
                      key="password"
                    >
                      password
                    </option>
                    <option
                      key="string"
                    >
                      string
                    </option>
                  </select>
                </div>
                <Connect(Options)
                  conf_spec_id="00000000-0000-0000-0000-000000000002"
                  options={
                    Array [
                      Object {
                        "id": "00000000-0000-0000-0000-000000000001",
                        "label": "https://example.com/",
                        "value": "https://example.com/",
                      },
                    ]
                  }
                >
                  <Options
                    addOptions={[Function]}
                    conf_spec_id="00000000-0000-0000-0000-000000000002"
                    options={
                      Array [
                        Object {
                          "id": "00000000-0000-0000-0000-000000000001",
                          "label": "https://example.com/",
                          "value": "https://example.com/",
                        },
                      ]
                    }
                  >
                    <div>
                      <div
                        className="confSpecRow"
                      >
                        <div
                          className="options-title"
                        >
                          Options
                          <button
                            className="addNewButton"
                            onClick={[Function]}
                            type="button"
                          >
                            + Add
                          </button>
                        </div>
                      </div>
                      <Connect(OptionsItem)
                        conf_spec_id="00000000-0000-0000-0000-000000000002"
                        key="00000000-0000-0000-0000-000000000001"
                        option={
                          Object {
                            "id": "00000000-0000-0000-0000-000000000001",
                            "label": "https://example.com/",
                            "value": "https://example.com/",
                          }
                        }
                      >
                        <OptionsItem
                          conf_spec_id="00000000-0000-0000-0000-000000000002"
                          deleteOption={[Function]}
                          option={
                            Object {
                              "id": "00000000-0000-0000-0000-000000000001",
                              "label": "https://example.com/",
                              "value": "https://example.com/",
                            }
                          }
                          updateOption={[Function]}
                        >
                          <div
                            className="optionsWrapper"
                          >
                            <div
                              className="column"
                            >
                              <div
                                className="inputDiv"
                              >
                                value
                                <span
                                  className="required-field"
                                >
                                  *
                                </span>
                                <input
                                  className="customInput"
                                  name="value"
                                  onChange={[Function]}
                                  required={true}
                                  type="text"
                                  value="https://example.com/"
                                />
                              </div>
                              <div
                                className="inputDiv"
                              >
                                label
                                <span
                                  className="required-field"
                                >
                                  *
                                </span>
                                <input
                                  className="customInput"
                                  name="label"
                                  onChange={[Function]}
                                  required={true}
                                  type="text"
                                  value="https://example.com/"
                                />
                              </div>
                            </div>
                            <div
                              className="optionsIconWrapper"
                            >
                              <svg
                                className="closeIcon"
                                onClick={[Function]}
                              >
                                <use
                                  xlinkHref="sprite.svg#icon-small-x-close"
                                />
                              </svg>
                            </div>
                          </div>
                        </OptionsItem>
                      </Connect(OptionsItem)>
                    </div>
                  </Options>
                </Connect(Options)>
                <div
                  className="inputDiv"
                >
                  <label>
                    label
                  </label>
                  <span
                    className="required-field"
                  >
                    *
                  </span>
                  <input
                    className="customInput"
                    name="label"
                    onBlur={[Function]}
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value="URL"
                  />
                </div>
                <div
                  className="inputDiv"
                >
                  <label>
                    tooltip
                  </label>
                  <input
                    className="customInput"
                    name="tooltip"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value="The base URL of the Serverless Relay"
                  />
                </div>
                <div
                  className="inputDiv"
                >
                  <label>
                    subtype
                  </label>
                  <input
                    className="customInput"
                    name="subtype"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                  />
                </div>
                <div
                  className="inputDiv"
                >
                  <label>
                    group
                  </label>
                  <input
                    className="customInput"
                    name="group"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                  />
                </div>
                <div
                  className="checkboxDiv"
                >
                  <label>
                    required
                  </label>
                  <input
                    checked={true}
                    name="required"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </div>
              <div>
                <svg
                  className="closeIcon"
                  onClick={[Function]}
                >
                  <use
                    xlinkHref="sprite.svg#icon-small-x-close"
                  />
                </svg>
              </div>
            </div>
          </ConfigurationSpecItem>
        </Connect(ConfigurationSpecItem)>
      </div>
    </ConfigurationSpec>
  </Connect(ConfigurationSpec)>
</Provider>
`;

exports[`ConfigurationSpec should render ConfigurationSpec without ConfigurationSpecItems 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ConfigurationSpec)>
    <ConfigurationSpec
      addConfSpec={[Function]}
      syncConfSpec={Array []}
    >
      <div>
        <div
          className="confSpecRow"
        >
          <div>
            Configuration Spec
            <button
              onClick={[Function]}
              type="button"
            >
              + Add
            </button>
          </div>
        </div>
      </div>
    </ConfigurationSpec>
  </Connect(ConfigurationSpec)>
</Provider>
`;
